### ISSUE TITLE
Encoding issues on Ubuntu 14.04

### ISSUE THREAD

**Issue by OEP** [ScottDuckworth/python-anyvcs#51]
I just noticed a few of the encoding unit tests are failing on my 14.04 system.

```
pkilgo@vroom:~/projects/python-anyvcs$ TEST_LOG_FILE=testlog python -m unittest tests.SvnUTF8EncodingTest
/tmp/anyvcs-test.DdLGkr
E.E
======================================================================
ERROR: test_cat (tests.SvnUTF8EncodingTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests.py", line 2308, in test_cat
    result = self.repo.cat(self.main_branch, aenema)
  File "anyvcs/svn.py", line 278, in cat
    ls = self.ls(rev, path, directory=True)
  File "anyvcs/svn.py", line 250, in ls
    proplist = self._proplist(revstr, name)
  File "anyvcs/svn.py", line 123, in _proplist
    output = self._command(cmd).decode()
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 16: ordinal not in range(128)

======================================================================
ERROR: test_ls (tests.SvnUTF8EncodingTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests.py", line 2303, in test_ls
    result = self.repo.ls(self.main_branch, '/', report=['target'])
  File "anyvcs/svn.py", line 250, in ls
    proplist = self._proplist(revstr, name)
  File "anyvcs/svn.py", line 123, in _proplist
    output = self._command(cmd).decode()
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 16: ordinal not in range(128)

----------------------------------------------------------------------
Ran 3 tests in 0.583s

FAILED (errors=2)

pkilgo@vroom:~/projects/python-anyvcs$ cat testlog 
Checked out revision 0.
Adding         b
Adding         Ænema
Transmitting file data ..
Committed revision 1.
Updating '.':
At revision 1.
['svn', 'checkout', 'file:///tmp/anyvcs-test.DdLGkr/main', '/tmp/anyvcs-test.DdLGkr/work']
['svn', 'status', '--xml']
['svn', 'add', '--force', '-q', 'b']
['svn', 'add', '--force', '-q', u'\xc6nema']
['svn', 'commit', '-m', 'modify \xc3\x86nema']
['svn', 'update']
```

Looks like this command is the culprit:

```
['svnlook', 'proplist', '-r', '1', '.', u'/\xc6nema']
```

Versions:

```
pkilgo@vroom:~/projects/python-anyvcs$ python --version
Python 2.7.6

pkilgo@vroom:~/projects/python-anyvcs$ svn --version
svn, version 1.8.8 (r1568071)
   compiled Feb 24 2014, 10:11:25 on x86_64-pc-linux-gnu

Copyright (C) 2013 The Apache Software Foundation.
This software consists of contributions made by many people;
see the NOTICE file for more information.
Subversion is open source software, see http://subversion.apache.org/

The following repository access (RA) modules are available:

* ra_svn : Module for accessing a repository using the svn network protocol.
  - with Cyrus SASL authentication
  - handles 'svn' scheme
* ra_local : Module for accessing a repository on local disk.
  - handles 'file' scheme
* ra_serf : Module for accessing a repository via WebDAV protocol using serf.
  - using serf 1.3.3
  - handles 'http' scheme
  - handles 'https' scheme
```

**Comment by OEP** [comment]
Looks like there could be some minor headaches to deal with:

```
$> svnlook proplist -r 1 . /
$> svn --version --quiet
1.6.17
```

```
$> svnlook proplist -r 1 . /
Properties on '/':
$> svnlook --version --quiet
1.8.8
```

So I guess somewhere in between 1.7 and 1.8.8, SVN started listing the path in the first line of the output. Our unicode tests caught this because there is a bare `decode()` in `SvnRepo._proplist()`.

Looks like there are a few things we can do:
1. Strip the first line of output according to the version of SVN (or some heuristic to tell if the first line is not needed)
2. If we think there could be unicode characters in SVN props (probably) throw `self.encoding` into some of those `decode()`'s.

**Comment by OEP** [comment]
A bisect suggests this started in 1.8.0

**Comment by OEP** [comment]
merged 88f629e
